tests/unit/core/statistics/test_constraints.py

```diff
@@ -1264,6 +1267,314 @@ def test_unique_proportion_between_constraint_wrong_datatype():
         columnUniqueValueProportionBetweenConstraint(lower_fraction=0.4, upper_fraction=2)
 
 
+def test_table_shape_constraints(df_lending_club, local_config_path):
+
+    rows = numberOfRowsConstraint(n_rows=10)
+    rows_2 = numberOfRowsConstraint(n_rows=len(df_lending_club.index))
+
+    column_exist = columnExistsConstraint("no_WAY")
+    column_exist2 = columnExistsConstraint("loan_amnt")
+
+    set1 = set(["col1", "col2"])
+    set2 = set(df_lending_club.columns)
+    columns_match = columnsMatchSetConstraint(set1)
+    columns_match2 = columnsMatchSetConstraint(set2)
+
+    table_shape_constraints = [rows, rows_2, column_exist, column_exist2, columns_match, columns_match2]
+
+    dc = DatasetConstraints(None, table_shape_constraints=table_shape_constraints)
+
+    config = load_config(local_config_path)
+    session = session_from_config(config)
+
+    profile = session.log_dataframe(df_lending_club, "test.data", constraints=dc)
+
+    report = profile.apply_table_shape_constraints()
+
+    assert len(report) == 6
+    # table shape {self.value} {Op.Name(self.op)} {self.first_field}
+    assert report[0][0] == f"table total_row_number {Op.Name(Op.EQ)} 10"
+    assert report[0][1] == 1
+    assert report[0][2] == 1
+
+    assert report[1][0] == f"table total_row_number {Op.Name(Op.EQ)} {len(df_lending_club.index)}"
+    assert report[1][1] == 1
+    assert report[1][2] == 0
+
+    assert report[2][0] == f"table columns {Op.Name(Op.CONTAIN)} no_WAY"
+    assert report[2][1] == 1
+    assert report[2][2] == 1
+
+    assert report[3][0] == f"table columns {Op.Name(Op.CONTAIN)} loan_amnt"
+    assert report[3][1] == 1
+    assert report[3][2] == 0
+
+    assert report[4][0] == f"table columns {Op.Name(Op.EQ)} {set1}"
+    assert report[4][1] == 1
+    assert report[4][2] == 1
+
+    reference_set_str = ""
+    if len(set2) > 20:
+        tmp_set = set(list(set2)[:20])
+        reference_set_str = f"{str(tmp_set)[:-1]}, ...}}"
+    else:
+        reference_set_str = str(set2)
+    assert report[5][0] == f"table columns {Op.Name(Op.EQ)} {reference_set_str}"
+    assert report[5][1] == 1
+    assert report[5][2] == 0
+
+
```

Comment:
parametrize?

Reply:
Yeah it is a good suggestion. Ok to do this in a follow up change.